homeassistant/components/zha/sensor.py

```diff
@@ -1249,3 +1265,45 @@ class SonoffPresenceSenorIlluminationStatus(Sensor):
     def formatter(self, value: int) -> int | float | None:
         """Numeric pass-through formatter."""
         return SonoffIlluminationStates(value).name
+
+
+@CONFIG_DIAGNOSTIC_MATCH(cluster_handler_names=CLUSTER_HANDLER_THERMOSTAT)
+# pylint: disable-next=hass-invalid-inheritance # needs fixing
+class PiHeatingDemand(Sensor):
```

Comment:
uh, I'll need to have another look why changing from multi match to a diagnostic match makes this uncovered.

Reply:
It looks a little bit like a bug to me in the coverage calculation, because it seems to think an empty line is not covered. I'll just try to rebase before anything else.

Reply:
Looks better now.